pyscf/solvent/smd.py

```diff
@@ -0,0 +1,631 @@
+# Copyright 2023 The GPU4PySCF Authors. All Rights Reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+SMD solvent model, copied from GPU4PYSCF with modification for CPU
+'''
+
+import numpy as np
+import scipy
+from pyscf import lib
+from pyscf.data import radii
+from pyscf.dft import gen_grid
+from pyscf.solvent import pcm, _attach_solvent
+from pyscf.lib import logger
+
+@lib.with_doc(_attach_solvent._for_scf.__doc__)
+def smd_for_scf(mf, solvent_obj=None, dm=None):
+    if solvent_obj is None:
+        solvent_obj = SMD(mf.mol)
+    return _attach_solvent._for_scf(mf, solvent_obj, dm)
+
+# Inject PCM to SCF, TODO: add it to other methods later
+from pyscf import scf
+scf.hf.RHF.SMD = smd_for_scf
+scf.uhf.UHF.SMD = smd_for_scf
+hartree2kcal = 627.5
+# see https://pubs.acs.org/doi/epdf/10.1021/jp810292n
+
+sigma_water = {
+    ('H'): 48.69,
```

Comment:
1. Consider to use only one type as the key, like ('H',) .
2. Does it make sense to explicitly define the parameters for all possible bonding combinations? Then replacing `sigma_xxx.get(bond, 0.0)` by `sigma_xxx[bond]` . This can help to identify typo or any accidental errors.

Reply:
I'm not sure if we want to update this. These parameters are in the fortran code now. They are still here in case anyone want to use the code to do some research. Although a lot of unit tests and benchmarks have been done for the python code, to be honest, I am not sure if all the cases are covered.